### PR TITLE
Implement CMake bracket comment syntax

### LINKED ISSUE #2338: Add support for CMake bracket comment
It would be nice to support CMake bracket comment.
A bracket comment is a new style for comments introduced in CMake 3.0 that can be used for multiline comments. It's inspired by Lua opening long bracket (already supported by Pygments).

`# inline CMake comment`

```
#[[ multiline
CMake comment
]]
```

```
#[==[ another multiline
CMake comment
]==]
```
See documentation for  [comments](https://cmake.org/cmake/help/git-master/manual/cmake-language.7.html#comments) and [bracket argument](https://cmake.org/cmake/help/git-master/manual/cmake-language.7.html#bracket-argument) for details.

### PR DESCRIPTION
**Pull request by rvanvelzen**
Closes #2338

### CODE REVIEW

File: pygments/lexers/make.py [hunk 1]
```diff
@@ -193,6 +193,7 @@ class CMakeLexer(RegexLexer):
         ],
         'ws': [
             (r'[ \t]+', Whitespace),
+            (r'(?:#\[(?P<level>=*)\[[\w\W]*?\](?P=level)\])', Comment),
```
Comment (birkenfeld) [inline pygments/lexers/make.py:196]: The outer group `(?:...)` shouldn't be necessary here...

**Review by birkenfeld — COMMENTED** [review]
Thanks, looks good except for a small question.

File: pygments/lexers/make.py (hunk 1 above)
Comment (rvanvelzen) [reply pygments/lexers/make.py]: Thanks, adjusted

**Comment by birkenfeld** [comment]
Thanks!

**Pull request merged**